lib/membrane/mpeg/ts/demuxer.ex

```diff
@@ -38,13 +49,14 @@ defmodule Membrane.MPEG.TS.Demuxer do
   end
 
   @impl true
-  def handle_pad_added(pad = {Membrane.Pad, _, {:stream_id, sid}}, _, state) do
+  def handle_pad_added(pad = {Membrane.Pad, _, {:stream_id, sid}}, ctx, state) do
+    aligned? = ctx.options[:aligned?]
     format =
-      case Map.fetch!(state.demuxer.pmt.streams, sid) do
-        %{stream_type: :H264} -> %Membrane.H264{alignment: :nalu}
-        _ -> %Membrane.RemoteStream{}
+      cond do
+        Map.fetch!(state.demuxer.pmt.streams, sid)[:stream_type] == :H264  and aligned? -> %Membrane.H264{alignment: :au}
```

Comment:
The change from `alignment: :nalu` to `alignment: :nalu` is dictated by the fact that as far as I know the alignment in context of H264 elementary stream should be considered in respect to frames (access units)